akd_local_auditor/src/storage/s3/mod.rs

```diff
@@ -178,14 +178,68 @@ impl S3AuditStorage {
             .into_paginator()
     }
 
-    async fn get_object(&self, key: &str) -> s3::client::fluent_builders::GetObject {
-        let config = self.get_config().await;
+    async fn get_object(&self, key: &str) -> Result<s3::client::fluent_builders::GetObject> {
+        // We need to retrieve the minimum version of an object, to guarantee we see the ORIGINAL
+        // status of the blob. Future versions are not supported
 
-        s3::Client::from_conf(config)
+        let attributes = s3::Client::from_conf(self.get_config().await)
+            .list_object_versions()
+            .bucket(self.bucket.clone())
+            .prefix(key.to_string())
+            .max_keys(1)
+            .send()
+            .await?;
+
+        let mut version_count = 0usize;
+        let mut min_time = std::time::SystemTime::now();
```

Comment:
We shouldn't assume clock synchrony here between the device this code is running on and the S3 modification times. Why not use only the modification times to calculate the min?

Reply:
So that's exactly what we're using, however (unfortunately) the AWS SDK's implementation of `DateTime` doesn't support comparisons. So we convert it to a local `SystemTime` to do the comparisons. Now I understand, if we were trying to compare against a local system time, we'd have a problem with clock drift, however we're converting ALL of the aws `DateTime`s to `SystemTime` to the relative differences should be the same (i.e. identifying the minimum time shouldn't be impacted).

There's many quirks in the AWS SDK...